### PR TITLE
[xgboost/release] Xgboost/connect gpu test

### PR DESCRIPTION
**Pull request by krfricke**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by krfricke** [comment]
Passed: https://buildkite.com/ray-project/periodic-ci/builds/1422#_

File: release/xgboost_tests/app_config_gpu.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ post_build_cmds:
   - sudo rm -rf /home/ray/anaconda3/lib/python3.7/site-packages/numpy || true
   - pip3 install numpy || true
   - pip3 install -U {{ env["RAY_WHEELS"] | default("ray") }}
+  - pip3 install -U "xgboost_ray" petastorm  # Install latest releases
```
Comment (amogkam) [inline release/xgboost_tests/app_config_gpu.yaml:18]: I think we want to be testing using xgboost_ray master right?

**Review by amogkam — COMMENTED** [review]
Thanks, LGTM overall! But I think we want to be testing using xgboost-ray master and not latest, right? If we treat xgboost-ray as an extension of ray, then on the OSS side we should be testing with ray master against latest anyscale.

File: release/xgboost_tests/app_config_gpu.yaml (hunk 1 above)
Comment (krfricke) [reply release/xgboost_tests/app_config_gpu.yaml:18]: Do we? I thought users would usually just use the latest release in most cases (and release circles are very quick for these libraries).

Should we maybe just test both?

**Comment by krfricke** [comment]
Both tests pass: https://buildkite.com/ray-project/periodic-ci/builds/1443

**Review by amogkam — APPROVED** [review]

**Pull request merged**